pkg/ansible/proxy/proxy.go

```diff
@@ -210,6 +213,53 @@ func addWatchToController(owner kubeconfig.NamespacedOwnerReference, cMap *contr
 	awMap := contents.AnnotationWatchMap
 	u := &unstructured.Unstructured{}
 	u.SetGroupVersionKind(ownerMapping.GroupVersionKind)
+	
+	//Adding dependentPredicate for avoiding reconciles when dependent objects are not changed by user (borrowed from Helm operator)
+	dependentPredicate := predicate.Funcs{
+		// We don't need to reconcile dependent resource creation events
+		// because dependent resources are only ever created during
+		// reconciliation. Another reconcile would be redundant.
+		CreateFunc: func(e event.CreateEvent) bool {
+			o := e.Object.(*unstructured.Unstructured)
+			log.V(1).Info("Skipping reconciliation for dependent resource creation", "name", o.GetName(), "namespace", o.GetNamespace(), "apiVersion", o.GroupVersionKind().GroupVersion(), "kind", o.GroupVersionKind().Kind)
+			return false
+		},
+
+		// Reconcile when a dependent resource is deleted so that it can be
+		// recreated.
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			o := e.Object.(*unstructured.Unstructured)
+			log.V(1).Info("Reconciling due to dependent resource deletion", "name", o.GetName(), "namespace", o.GetNamespace(), "apiVersion", o.GroupVersionKind().GroupVersion(), "kind", o.GroupVersionKind().Kind)
+			return true
+		},
+
+		// Don't reconcile when a generic event is received for a dependent
+		GenericFunc: func(e event.GenericEvent) bool {
```

Comment:
Hi @joelanford 

Initially we added this along with event filtering for update event. i think we can keep it, just in case we do get generic events from the API server. That said, haven't see logs related to generic events coming in from the API server (at least for OCP 3.11 we are currently using)

Reply:
Makes sense. Thanks! In that case, I think its okay for this to be included in the predicate used by the helm operator. These are otherwise identical right? If so, let's keep the `GenericFunc` and refactor so that this can be imported and used by both ansible and helm operators. WDYT?

Reply:
Yes, other than generic event handling and the option to disable predicates, from watches.yaml, both are same. Will update the code as suggested. Please let me know if we still want to use the option to enable/disable predicates for ansible operator.